src/wovnio/html/HtmlConverter.php

```diff
@@ -45,98 +44,89 @@ public function __construct($html, $encoding, $token, $store = null, $headers =
    *
    * @return array converted html and HtmlReplaceMarker
    */
-  public function convertToAppropriateForApiBody($removeParts = true)
+  public function insertSnippetAndHreflangTags()
   {
-    if ($this->encoding && in_array($this->encoding, self::$supported_encodings)) {
-      $encoding = $this->encoding;
-    } else {
-      // Encoding detection uses 30% of execution time for this method.
-      $encoding = mb_detect_encoding($this->html, self::$supported_encodings);
-    }
-
-    $dom = simple_html_dom::str_get_html($this->html, $encoding, false, false, $encoding, false);
-    $this->insertSnippet($dom);
-    if (isset($this->store) && isset($this->headers)) {
-      $this->insertHreflangTags($dom);
-    }
-
-    $marker = new HtmlReplaceMarker();
-    if ($removeParts) {
-      $this->removeWovnIgnore($dom, $marker);
-      $this->removeForm($dom, $marker);
-      $this->removeScript($dom, $marker);
-    }
-
-    $converted_html = $dom->save();
-
-    // Without clear(), Segmentation fault will be raised.
-    // @see https://sourceforge.net/p/simplehtmldom/bugs/103/
-    $dom->clear();
-    unset($dom);
+    $this->html = $this->insertSnippet($this->html);
+    $this->html = $this->insertHreflangTags($this->html);
 
-    if ($removeParts) {
-      $converted_html = $this->removeBackendWovnIgnoreComment($converted_html, $marker);
-    }
-    return array($converted_html, $marker);
+    return array($this->html);
   }
 
   /**
    * Insert wovn's snippet to ensure snippet is always inserted.
    * When snippet is always inserted, do nothing
    *
-   * @param simple_html_dom_node $dom
+   * @param string $html
    */
-  private function insertSnippet($dom)
+  private function insertSnippet($html)
   {
-    foreach ($dom->find('script') as $node) {
-      if (strpos($node->getAttribute('src'), '//j.wovn.io/') !== false ||
-        strpos($node->getAttribute('src'), '//j.dev-wovn.io:3000/') !== false) {
-        return;
-      }
+    $snippet_regex = "<\<script(.*)src='(j.wovn.io|j.dev-wovn.io)'(.*)><\/script>";
```

Comment:
Does this regex match the standard WOVN.io snippet in your environment? From looking at it I think it's wrong, I'm working on a correct version.
sample snippet:
`<script src="//j.wovn.io/1" data-wovnio="key=hS7IIB" async></script>`

Reply:
This regex should match all environments of WOVN, and also it matches the standard widget.

`$snippet_regex = "<script(.*)src=[^ ]*(j.[^.]*wovn.io)(.*)><\/script>";`

Reply:
Thank you. I changed your regex a bit. Please check again.